### DISCUSSION TITLE
Honestly: I hate git

### DISCUSSION THREAD

**Discussion by BerndStorck** [community/community#23675] [New to GitHub]
<pre><code class="lang-auto">                       Sequel 348

                 What happened until now
</code></pre>
<p>I have written a small script. Because of trouble using git because I was not able to upload the offline files to the new empty online repository I have uploaded the files via the Github page last night in the online new created repository.</p>
<p>Now <strong>I fail to add / upload changes I’ve made offline</strong>.</p>
<p>Currently I have no clue how to handle it. I’ve tried “git rebase” and now “git init”.</p>
<pre><code class="lang-auto">            Will Bernd solve this issue in this sequel
               or will he leave this project for ever?

                                Stay tuned!
</code></pre>
<p><em>Bernd in his coding cabinet staring at a pale green glowing screen, mumbeling:</em></p>
<p>Better than trying to understand that git / Github mess currently seems for me</p>
<ol>
<li>to rename my offline folder,</li>
<li>clone the online repository,</li>
<li>copy the changed files in the new folder with the cloned online repository,</li>
<li>add,</li>
<li>commit and</li>
<li>push them.</li>
</ol>
<p>Git and Githib do make more stress than coding a MB application. It’s a mess, absolutly overcomplicated.</p>
<p><em>Loudly finally he says:</em> <strong>How can I upload the changed files with git to Github?</strong> <em>Mumbling again:</em> And where is a caffeine drink?</p>

**Comment by kingthorin** [comment]
<p>You shot yourself in the foot by re-initializing the local copy of the repo.</p>
<p>Your six steps seem fine.</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/berndstorck">@BerndStorck</a>!</p>
<p>I feel you. Git honestly has a pretty steep learning curve which can be extremely frustrating. Once you get your workflow down though, it gets a lot easier.</p>
<p>As <a class="mention" href="/kingthorin">@kingthorin</a> said, your steps for getting back into the flow seem good. There are other ways to troubleshoot issues and such but sometimes just starting clean from scratch with your local repository is just the easiest way of going about things.</p>
<p>When you need to sync your local repository with your remote repository before making changes to your code, run <a href="https://git-scm.com/docs/git-fetch"><strong>git fetch</strong></a> which updates your branches.</p>
<p>Have you taken a look at the <a href="https://guides.github.com/">GitHub Guides</a>? The <strong>Understanding the GitHub Flow</strong> guide and the <strong>Hello World</strong> guide are ones that helped me a lot when I was first getting started learning Git. I think they do a decent job of breaking down the fundamental concepts.</p>
<p>Hope that helps!</p>

**Comment by BerndStorck** [comment]
<p>Hi <a class="mention" href="/that-pat">@that-pat</a>, hi <a class="mention" href="/kingthorin">@kingthorin</a>!</p>
<p><strong>Thanks for your both answers.</strong> :slight_smile:</p>
<p>I’ve done it with the 6 steps above, and it has worked.</p>
<p><a class="mention" href="/that-pat">@that-pat</a>:<br>
I have looked at some tutorials and some basic videos about git and Github. But every – even very good – introducing video I’ve seen at Youtube has shown only a special way and situation. If you are one step aside those special short shown use cases, you are lost because of missing knowledge.</p>
<p>I plan to buy and read a book about git, but in the last time my days are consumed by grooving my abilities and knowledge in programming Bash scripts. One of the last steps is sometimes to publish a script at Github after I’ve coded and tested the script for a while.  At this point I mostly need rather sleep than – in those moments to me – a little bit prissy appearing tool named git. :wink:</p>
<p>I will look at the linked learning resources (again) and study them.</p>
<h2>Thanks for your friendly answers!</h2>
<p>For German speaking linux friends, it might be not bad to visit: <a href="https://www.facebook.com/BStLinux" rel="noopener nofollow ugc">https://www.facebook.com/BStLinux</a> There I discuss and share issues about Linux and Bash Scripting.</p>

**Comment by endolith** [comment]
<p>I just use SmartGit.  I only deal with the command line version for occasional weird cases that it can’t handle.</p>

**Comment by bskidan** [comment]
<p>I’d recommend reading this free book: <a href="https://github.com/pluralsight/git-internals-pdf" rel="noopener nofollow ugc">https://github.com/pluralsight/git-internals-pdf</a></p>
<p>For me, that was exactly what took all the guessing away. I understand git now from the inside and what to do, when and how. Short and easy read.</p>

**Comment by mikernet** [comment]
<p>GitKraken is by far the best Git client I have used, I highly recommend it. I haven’t used the Git command line in forever. It has made onboarding new devs with little to no Git experience much easier.</p>

**Comment by Instrumcentax15** [comment]
I can't commit from Xcode. It asks for an ssh key. I followed instructions on generating one and still "operation failed". There is no way to do a search on Github's home page because once you have an account, just like Facebook, it searches only in terms of your profile.

**Comment by osirisgothra** [comment]
what ticks me off is I would be expected  to (for the third time) spend precious time re-learning something that had already been set up perfectly (for the third time), just for doing one or two things. Things that couldn't care less about security because it is seriously doubtful that it's the next holy grail app. Ended up abandoning git and github altogether because of that and ended up just going back to using it locally and also a third party private http server. It is arguably insecure as heck but really unless someone hacks it for the 1 hour I'm upgrading my PC every so many years, it wouldn't matter anyway.  

Very little free time, I have. And I what I don't have is time to nitpick something I already invested my very little free time in learning what I needed to know and how to keep things going smoothly. But Its like everyone has to mess with things, be nice if we could at least opt out of the changes, give us a little legacy corner or something, but what am I saying, once microsoft gets their tentacles in something the ensh*tification soon follows.

 My advice is as soon as you have an alternate place to set things up, do it, you never know when they will upend the entire community (again) and cancel out your beloved workflow just to cater to the new majority that is sure to come along in the next decade. Actually the only sure thing is that it will happen, it's really not even a question of "if" (I mean really, its happened a couple times already in the past 20 or so years, why would it stop there?).
